redbeams/src/main/java/com/sequenceiq/redbeams/service/dbconfig/DatabaseConfigService.java

```diff
@@ -105,25 +117,42 @@ public DatabaseConfig delete(String name, String environmentId) {
     }
 
     private DatabaseConfig deleteOne(DatabaseConfig databaseConfig) {
+        LOGGER.debug("Deleting database with name: {}", databaseConfig.getName());
         try {
+            // FIXME this can take a while and trigger logging that the transaction took too long
             return transactionService.required(() -> {
-                delete(databaseConfig);
-                if (databaseConfig.isUserManaged()) {
-                    deletePhysicalDatabase(databaseConfig);
+                if (databaseConfig.getStatus() == ResourceStatus.SERVICE_MANAGED) {
+                    return deleteServiceManagedDatabase(databaseConfig);
+                } else {
+                    return delete(databaseConfig);
```

Comment:
I see that `deleteServiceManagedDatabase` does delete (archive) the database from table DatabaseConfig, but I think it would make sense to do the delete for both cases (user_managed / service_managed) at the same place.

Reply:
It's another valid control flow.